code/components/citizen-resources-core/include/StateBagComponent.h

```diff
@@ -97,8 +106,9 @@ class CRC_EXPORT StateBagComponent : public fwRefCountable, public IAttached<Res
 
 	//
 	// Should be called when receiving a state bag control packet.
+	// return an optional int of the entity id, it's only given if the packet was not processed and can be retried if required
 	//
-	virtual void HandlePacket(int source, std::string_view data) = 0;
+	virtual std::optional<uint32_t> HandlePacket(int source, std::string_view data) = 0;
```

Comment:
A bit ugly. I'd rather see a success/fail return value + an out pointer argument that would get (a move-copy of?) the bag name put there, if rejected for such reason.